pkg/cluster/update_worker_test.go

```diff
@@ -0,0 +1,114 @@
+package cluster
+
+import (
+	"context"
+	"crypto/rsa"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/cluster/updateblob"
+	"github.com/openshift/openshift-azure/pkg/util/mocks/mock_azureclient"
+	"github.com/openshift/openshift-azure/pkg/util/mocks/mock_cluster"
+	"github.com/openshift/openshift-azure/pkg/util/mocks/mock_kubeclient"
+	"github.com/openshift/openshift-azure/pkg/util/mocks/mock_scaler"
+	"github.com/openshift/openshift-azure/pkg/util/mocks/mock_updateblob"
+)
+
+func TestUpdateWorkerAgentPool(t *testing.T) {
+	tests := []struct {
+		name   string
+		cs     *api.OpenShiftManagedCluster
+		vms    []compute.VirtualMachineScaleSetVM
+		suffix string
+		want   *api.PluginError
+	}{
+		{
+			name:   "basic coverage",
+			suffix: "foo",
+			cs: &api.OpenShiftManagedCluster{
+				Config: api.Config{
+					SSHKey: &rsa.PrivateKey{},
+				},
+				Properties: api.Properties{
+					AgentPoolProfiles: []api.AgentPoolProfile{
+						{
+							Name: "compute",
+							Role: api.AgentPoolProfileRoleCompute,
+						},
+					},
+					AzProfile: api.AzProfile{
+						ResourceGroup: "resourcegroup",
+					},
+				},
+			},
+			vms: []compute.VirtualMachineScaleSetVM{
+				{
+					Name:       to.StringPtr("ss-compute_0"),
+					InstanceID: to.StringPtr("0"),
+					VirtualMachineScaleSetVMProperties: &compute.VirtualMachineScaleSetVMProperties{
+						OsProfile: &compute.OSProfile{
+							ComputerName: to.StringPtr("compute-000000"),
+						},
+					},
+				},
+				{
+					Name:       to.StringPtr("ss-compute_1"),
+					InstanceID: to.StringPtr("1"),
+					VirtualMachineScaleSetVMProperties: &compute.VirtualMachineScaleSetVMProperties{
+						OsProfile: &compute.OSProfile{
+							ComputerName: to.StringPtr("compute-000001"),
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gmc := gomock.NewController(t)
+			defer gmc.Finish()
+
+			ctx := context.Background()
+			log := logrus.NewEntry(logrus.StandardLogger())
+			ubs := mock_updateblob.NewMockBlobService(gmc)
+			vmc := mock_azureclient.NewMockVirtualMachineScaleSetVMsClient(gmc)
+			ssc := mock_azureclient.NewMockVirtualMachineScaleSetsClient(gmc)
+			kclient := mock_kubeclient.NewMockKubeclient(gmc)
+			hasher := mock_cluster.NewMockHasher(gmc)
+			ss := &compute.VirtualMachineScaleSet{
+				Name: to.StringPtr("ss-compute-1"),
+				Sku:  &compute.Sku{Capacity: to.Int64Ptr(0)}}
+
+			targetScaler := mock_scaler.NewMockScaler(gmc)
+			var count int64 = 0
+			targetScaler.EXPECT().Scale(ctx, count)
```

Comment:
could s/count/int64(0)/ here fwiw

Reply:
I'm a bit confused that ss is a scaleset with 0 capacity and the test is expecting a scale to 0.  I don't call that "basic coverage".

Reply:
agreed, i have a better test on the way